addons/hr_expense/static/src/views/expense_list_view.js

```diff
@@ -0,0 +1,49 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+
+import { ListRenderer } from "@web/views/list/list_renderer";
+import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
+
+import { useService } from "@web/core/utils/hooks";
+
+export class ExpenseListRenderer extends ListRenderer {
+    setup() {
+        this.threadService = useService("mail.thread");
+        super.setup();
+    }
+
+    /** @override **/
+    async onCellClicked(record, column, ev) {
+        const sheetId = record.data.sheet_id[0];
+        const sheetThread = this.threadService.getThread('hr.expense.sheet', sheetId);
+        const lineThread = this.threadService.getThread('hr.expense', record.resId)
+        const attachments = lineThread.attachments
+
+        if (attachments.length > 0) {
+            const latestAttachment = attachments[attachments.length - 1];
+            sheetThread.update({ mainAttachment: latestAttachment });
+        }
+    }
+}
+
+export class ExpenseListWidget extends X2ManyField {
+    static components = {
+        ...X2ManyField.components,
+        ListRenderer: ExpenseListRenderer
+    };
+
+    // Originally a one2many field, expense_line_ids was temporarily changed to many2many in XML using the 'widget'
+    // attribute for adding expense lines inside the sheet. Now, with our custom X2ManyField widget, we override
+    // the getter to maintain the same behavior.
```

Comment:
no need to mention what it was... keep it simple and move it inside the function
// this field is used like a many2many list - allowing the selection of existing expenses